JC-587 copyedit of connectors quickstart (#443)
redhat-developer/app-services-guides: docs/connectors/getting-started-connectors/README.adoc

Learn how to create and set up connectors in {product-long-connectors}.

[#introduction]
Welcome to the quick start for {product-long-connectors}. 

In this quick start, you learn how to create a source connector and sink connector and send data to and from {product-kafka}. 

A *source* connector allows you to send data from an external system to {product-kafka}. A *sink* connector allows you to send data from {product-kafka} to an external system. 

endif::[]

ifndef::qs[]
== Overview

{product-long-kafka} is a cloud service that simplifies the process of running Apache Kafka. Apache Kafka is an open-source, distributed, publish-subscribe messaging system for creating fault-tolerant, real-time data feeds.  

> You can use {product-connectors} to configure communication between {product-kafka} instances and external services and applications. {product-connectors} allow you to configure how data moves from one endpoint to another without writing code. 

Reviewer: {product-connectors} doesn't seem to be defined in the attributes.

> 

Reviewer: We don't display images in the in-product quick starts, so the text explaining the diagram should be conditionalized to not be displayed in the quick start.

Also, in the line above the image, add the `[.screencapture]` metadata.

Author: That entire overview section is conditionalized to not display in the quick start - including the image

do I use [.screencapture] even if the image is not a screen capture but rather a diagram?

The following diagram illustrates how data flows from a source of data through a data source connector to a Kafka topic. And how data flows from a kafka topic to a  data sink connector to a data sink.

> image::connectors-diagram.png[Illustration of data flow from data source through Kafka to data sink]

Reviewer: Suggest adding a figure title to the diagram.

endif::[]

[id="proc-configuring-kafka-for-connectors_{context}"]
== Configuring the {product-kafka} instance for use with {product-long-connectors}

[role="_abstract"]
Configure your {product-kafka} for use with {product-long-connectors} by creating  Kafka topics, creating service accounts, and setting up access rules for the service accounts. The number of Kafka topics, service accounts that you create, and the access rules for the service accounts depend on your application. 

Kakfa topics store messages sent by producers (data sources) and make them available to consumers (data sinks). Service accounts allow you to connect and authenticate your Connectors with Kafka instances. Access rules for service accounts define how your Connectors accounts can access and use the associated Kafka instance and topics.

For this quick start, you create one Kafka topic, named *test*, one service account, and you provide all permissions on the service account.

ifndef::qs[]
.Prerequisites

> * You're logged in to the {service-url-connectors}[^] web console.

Reviewer: ```suggestion
* You're logged in to the web console at {service-url-connectors}[^] .
```

* You've created a  {product-kafka} instance and the instance is in the *Ready* state.

> For instructions on how to create a Kafka instance, see _Getting started with OpenShift Streams for Apache Kafka_. 

Reviewer: I think you can actually link to the Getting started quick start: "...see {base-url}{getting-started-url-kafka}[Getting started with {product-long-kafka}^].

. Create a service account for connectors: 
.. In the web console, select *Service Accounts*, and then click *Create service account*.
.. Type a unique service account name (for example, *test-service-acct* ) and then click *Create*.
.. Copy the generated *Client ID* and *Client Secret* to a secure location. You'll use these credentials to configure connections to this service account.
.. Select the *I have copied the client ID and secret* option, and then click *Close*.

. Set the level of access for your new service account in the Access Control List (ACL) of the Kafka instance:
.. Select *Streams for Apache Kafka* > *Kafka Instances*, click the name of the Kafka instance that you want the service account to access.
.. Click the *Access* tab to view the current ACL for this instance and then click *Manage access*.
.. From the *Account* drop-down menu, select the service account that you previously created, and then click *Next*.
.. Under *Assign Permissions*, use the drop-down menu to select the *Consume from a topic* and the *Produce to a topic* permission options, and then set all resource identifiers and values to `is "*"`. 
+

> The `is "*"` settings mean that the service account has rights to produce and receive messages for the Kafka topic that you created for Connectors. 

Reviewer: ```suggestion
The `is "*"` settings mean that the service account has permission to produce and receive messages for the Kafka topic that you created for Connectors. 
```

> .Verification

Reviewer: This is causing the "Verification" title to show up in the non-quickstart version. I'd suggest re-wording the qs verification to be something more tangible that the user can verify, and adding a "ifndef::qs[]" for the non-quickstart version.

For example, instead of "Have you completed these steps?", maybe do something like this:

```
.Verification
ifdef::qs[]
* Did you create a topic for connectors?
* Did you create a service account and save the credentials to a secure location?
* Did you set the *Consume from a topic* and *Produce to a topic* permissions for the service account?
endif::[]
ifndef::qs[]:
* Verify that the topic you created exists on the "Topics" page (or whatever the page is called).
* Verify that the service account credentials are copied to a secure location.
* Verify that the *Consume from a topic* and *Produce to a topic* permissions are set for the service account.
endif::[]
```
Alternatively, you could just not include any verification for this procedure.

ifdef::qs[]
* Have you completed these steps?
endif::[]

[id="proc-creating-source-connector_{context}"]
== Creating a Connectors instance for a data source

[role="_abstract"]
A source connector consumes events from an external data source and produces Kafka messages. For this quick start, you create an instance of the Data Generator Source connector. You configure the connector to listen for events from the data source and produce a Kafka message with a configurable payload for each event. The connector sends the messages at regular intervals to a Kafka topic on your {product-kafka} instance.

ifndef::qs[]
.Prerequisites
* You're logged in to the {service-url-connectors}[^] web console.

> * You configured a Kafka instance for Connectors as described in _Configuring the {product-kafka} instance for use with {product-long-connectors}_.

Reviewer: Add a link to this procedure.

.. Type the *Client ID* and *Client Secret* of the service account that you created for Connectors and then click *Next*.

. Provide connector-specific configuration:
.. *Data shape Format*: Accept the default, `application/octet-stream`.
.. *Topic Names*: Type the name of the topic that you created for Connectors (for the quick start example, type *test-topic*).
.. *Content Type*: Accept the default, `text/plain`.
.. *Message*: Type the content of the message that you want to send to the topic, for the quick start example, type `Hello World!`.
.. *Period*: Specify the interval (in milliseconds) at which you want the Connectors instance to sends messages to the Kafka topic. For the quick start example, specify `10000`, to send a message every 10 seconds.

. Optionally, configure the error handling policy for your Connectors instance. For the quick start, select *log* (the Connectors instance sends errors to its log).  +
+
Other options are *stop* (the Connectors instance shuts down in case of errors), or *dead letter queue* (the Connectors instance sends messages that it cannot handle to a dead letter topic that you define for the Connectors Kafka instance). 

> .. Click *Next*.

Reviewer: Should this and the following step be sub-steps of the preceding optional step, or are either/both actually regular steps?

.. Review the summary of the configuration properties of your Connectors instance and then click *Create Connectors instance* to deploy it.
+
Your Connectors instance is listed in the table of Connectors. After a couple of seconds, the status of your Connectors instance changes to the *Ready* state and it starts producing messages and sending them to its associated Kafka topic.
+
From the connectors table, you can stop, start and delete your Connectors instance, as well as edit its configuration by clicking the options icon (three vertical dots).
+
In the next procedure, you can verify that the source Connectors instance is sending messages as expected by creating a sink Connectors instance that consumes the messages.

> .Verification

Reviewer: Same as last procedure: make the verification a bit more specific about what exactly the user should verify, and add a non-quick start verification.

ifdef::qs[]
* Have you completed these steps?
endif::[]

[id="proc-creating-sink-connector_{context}"]
== Creating a Connectors instance for a data sink

[role="_abstract"]
A sink connector consumes messages from a Kafka topic and sends them to an external system. In this quick start, you use the *HTTP Sink* connector which consumes the Kakfa messages (produced by the source Connectors instance) and sends the message payloads to an HTTP endpoint.

ifndef::qs[]
.Prerequisites

> * You're logged in to the {service-url-connectors}[^] web console.

Reviewer: ```suggestion
* You're logged in to the web console at {service-url-connectors}[^] .
```

> * You created the source Connectors instance as described in _Creating a Connectors instance for a data source_).

Reviewer: Link to the procedure.

> * You have a unique URL from the https://webhook.site[webhook.site].

Reviewer: Would users know what this is and how to get it? It's the first time it's mentioned in the quick start.

Author: it is a prerq in the YAML file. But I also need to add a conditional prereq here for the user guide - thanks!

Author: actually it is there - as a prereq in the user guide :)

.. Click the *HTTP Sink connector* card and then click *Next*. 

. Select the {product-kafka} instance for the connector to work with. For the quick start, select *test*  and then click *Next*.

. Select the namespace to host your Connectors instance and then click *Next*.

. Configure the core configuration for your connector:
.. Provide a unique name for the connector. 
.. Type the *Client ID* and *Client Secret* of the service account that you created for Connectors and then click *Next*.

. Provide connector-specific configuration:
.. *Data shape Format*: Accept the default, `application/octet-stream`.
.. *Method*: Accept the default, `POST`.
.. *URL*: Enter your unique URL from link:https://webhook.site[webhook.site^].
.. *Topic Names*: Type the name of the topic that you used for the source Connectors instance (for the quick start example, type *test-topic*).

. Optionally, configure the error handling policy for your Connectors instance. For the quick start, select *log*  and then click *Next*.

> .. Review the summary of the configuration properties of your Connectors instance and then click *Create Connectors instance* to deploy it.

Reviewer: Is this supposed to be a sub-step?

+

> Your Connectors instance is listed in the table of Connectors. After a couple of seconds, the status of your Connectors instance changes to the *Ready* state and it consuming messages from the associated Kafka topic and sends them to the data sink (for the quick start, the data sink is the HTTP url that you provided).

Reviewer: ```suggestion
Your Connectors instance is listed in the table of Connectors. After a couple of seconds, the status of your Connectors instance changes to the *Ready* state and it starts consuming messages from the associated Kafka topic and sending them to the data sink (for the quick start, the data sink is the HTTP url that you provided).
```

> . Open the browser tab to link:https://webhook.site[webhook.site^] to see the HTTP POST calls with the message contents defined in the source connector.

Reviewer: ```suggestion
. Open a new web browser tab and navigate to link:https://webhook.site[webhook.site^] to see the HTTP POST calls with the message contents defined in the source connector.
```

Author: it needs to be the same browser tab - ot I have to reword this - will add a comment to deal with it later

// what are example messages?

> .Verification

Reviewer: Like before, add more specific verification.
